scripts/wdl/pon_gatk_workflow.wdl

```diff
@@ -0,0 +1,358 @@
+    }
+}
+
+# Calculate the target proportional coverage
+task CalculateTargetCoverage {
+    String entity_id
+    File padded_target_bed
+    String transform = "PCOV"
+    String grouping = "SAMPLE"
+    Boolean keep_duplicate_reads = true
+    Boolean disable_all_read_filters = false
+    File input_bam
+    File bam_idx
+    File ref_fasta
+    File ref_fasta_fai
+    File ref_fasta_dict
+    File gatk_jar
+    String is_disable_reference_validation
+    Boolean isWGS
+    Int mem
+
+    # Note that when isWGS is true, this task is still called by the workflow.
+    # In that case, to satisfy the output requirements an empty coverage file is created
+    # and passed to the WholeGenomeCoverage task.
+    command {
+        if [ ${isWGS} = false ]; \
+          then java -Xmx${mem}g -jar ${gatk_jar} CalculateTargetCoverage --output ${entity_id}.coverage.tsv \
+                --groupBy ${grouping} --transform ${transform} --targets ${padded_target_bed} --targetInformationColumns FULL \
+                --keepduplicatereads ${keep_duplicate_reads} --input ${input_bam} --reference ${ref_fasta} \
+                --disable_all_read_filters ${disable_all_read_filters} --interval_set_rule UNION --interval_padding 0 \
+                --secondsBetweenProgressUpdates 10.0 --disableSequenceDictionaryValidation ${is_disable_reference_validation} \
+                --createOutputBamIndex true --help false --version false --verbosity INFO --QUIET false; \
+          else touch ${entity_id}.coverage.tsv; \
+        fi
+    }
+
+    output {
+        File gatk_coverage_file = "${entity_id}.coverage.tsv"
+    }
+
+    #runtime {
+    #    docker: "gatk-protected/a1"
+    #}
+}
+
+# Caclulate coverage on Whole Genome Sequence using Spark. This task automatically creates
+# an output file, containing targets
+task WholeGenomeCoverage {
+    String entity_id
+    File coverage_file 
+    File target_file
+    File input_bam
+    File bam_idx
+    File ref_fasta
+    File ref_fasta_fai
+    File ref_fasta_dict
+    File gatk_jar
+    Boolean isWGS
+    Int wgsBinSize
+    Int mem
+
+    # If isWGS is set to true, the task produces WGS coverage and targets that are passed to downstream tasks
+    # If not, coverage and target file (received from upstream) for exome capture sequences are passed downstream
+    command {
+        if [ ${isWGS} = true ]; \
+          then java -Xmx${mem}g -jar ${gatk_jar} SparkGenomeReadCounts --outputFile ${entity_id}.coverage.tsv \
+                --reference ${ref_fasta} --input ${input_bam} --binsize ${wgsBinSize}; \
+          else ln -s ${coverage_file} ${entity_id}.coverage.tsv; ln -s ${target_file} ${entity_id}.coverage.tsv.targets.tsv; \
+        fi
+    }
+
+    output {
+        File gatk_coverage_file = "${entity_id}.coverage.tsv"
+        File gatk_target_file = "${entity_id}.coverage.tsv.targets.tsv"
+    }
+}
+
+# Helper task to aggregate array of targets, that scatter block outputs, into a single file
```

Comment:
I'm a little confused, but I think it's different from what I was trying to say here...